src/main/scala/cognite/spark/v1/DefaultSource.scala

```diff
@@ -259,6 +319,9 @@ object DefaultSource {
     val auth = apiKey
       .orElse(bearerToken)
       .getOrElse(sys.error("Either apiKey or bearerToken is required."))
+    val functionsAuth = parameters
+      .get("functionsApiKey")
```

Comment:
Why do we need a separate one for this? Is it just for testing?

Reply:
The functions are added in DefaultSource. During the initialization there we only have access to one apiKey which might not have permissions for functions. For example, if we read from publicdata, then we need a separate key to read from functions.

Reply:
Hmm, I see, that can be useful but then I'd suggest we use `"functionsApiKey"` if set, otherwise just use `"apiKey"`, as most users won't need to use a separate key for this and having to specify the same one twice would be annoying.